src/hooks/useClickOutside.ts

```diff
@@ -0,0 +1,33 @@
+import { RefObject, useEffect } from "react";
+
+interface ClickOutsideProps {
+  ref: RefObject<HTMLDivElement>;
+  buttonRef: RefObject<HTMLButtonElement>;
+  hide?: () => void;
+  closeRoot?: boolean;
+}
+
+export const useClickOutside = ({
+  ref,
+  hide,
+  closeRoot,
+  buttonRef,
+}: ClickOutsideProps) => {
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        ref.current &&
+        !ref.current.contains(event.target as Node) &&
+        buttonRef.current &&
+        !buttonRef.current.contains(event.target as Node)
+      ) {
+        hide && closeRoot && hide();
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
```

Comment:
Usually `mouseup` or `click` work better. The former gives the option of "canceling" the close by releasing the mouse elsewhere, but especially the latter since it's a higher level event than `down` or `up`

Reply:
ead67ab used `click`